src/partition/repart.c

```diff
@@ -1480,6 +1575,27 @@ static int context_read_definitions(
                 context->n_partitions++;
         }
 
+        /* Check that each configured verity hash/data partition has a matching verity data/hash partition. */
+
+        LIST_FOREACH(partitions, p, context->partitions) {
+                Partition *q, *s;
+
+                if (p->verity == VERITY_OFF)
+                        continue;
+
+                q = find_verity_sibling(context->partitions, p);
+                if (!q)
+                        return log_syntax(NULL, LOG_ERR, p->definition_path, 1, SYNTHETIC_ERRNO(EINVAL),
+                                          "Missing sibling partition for verity partition with VerityName=%s",
+                                          p->verity_name);
+
+                s = find_verity_sibling(q->partitions_next, p);
+                if (s)
+                        return log_syntax(NULL, LOG_ERR, p->definition_path, 1, SYNTHETIC_ERRNO(EINVAL),
+                                          "Multiple sibling partitions found for verity partition with VerityName=%s",
+                                          p->verity_name);
+        }
```

Comment:
i'd really hide the uniqueness check in find_verity_sibling() and make it return ENXIO if it can't find the sibling and ENOTUNIQ if not unique

And yeah, I think too it makes sense to cache the result in the partition structure (just add two pointer fields `verity_data_partition` and `verity_hash_partition` to it, no need to avoid wasting a few bytes here), and then use that for the actual verity operatoin instead of searching again